packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -352,6 +357,54 @@ export function makeVirtualObjectManager(
     configurable: true,
   });
 
+  /**
+   * Assess the facetiousness of a value.  If the value is an object containing
+   * only named properties and each such property's value is a function, `obj`
+   * represents a single facet and 'one' is returned.  If each property's value
+   * is instead an object of facetiousness 'one', `obj` represents multiple
+   * facets and 'many' is returned.  In all other cases `obj` does not represent
+   * any kind of facet abstraction and 'not' is returned.
+   *
+   * @typedef {'one'|'many'|'not'} Facetiousness
+   *
+   * @param {*} obj  The (alleged) object to be assessed
+   *
+   * @returns {Facetiousness} an assessment of the facetiousness of `obj`
+   */
+  function assessFacetiousness(obj) {
+    if (typeof obj !== 'object') {
+      return 'not';
+    }
+    if (Object.getOwnPropertySymbols(obj).length !== 0) {
+      return 'not';
+    }
+    // XXX should there be some kind of prototype check here for further paranoia?
+    let established;
+    for (const [_name, value] of Object.entries(obj)) {
+      let current;
+      if (typeof value === 'function') {
+        current = 'one';
+      } else if (
+        typeof value === 'object' &&
+        assessFacetiousness(value) === 'one'
+      ) {
+        current = 'many';
+      } else {
+        return 'not';
+      }
+      if (!established) {
+        established = current;
+      } else if (established !== current) {
+        return 'not';
+      }
+    }
+    if (!established) {
+      return 'one';
```

Comment:
So the empty object is interpreted as a single far object. Good. (No change suggested)